### ISSUE TITLE
Difference of Mobile OS usage(iOS, Android) - 4.1.2: Name, Role, Value

### ISSUE THREAD

**Issue by lgnacio** [w3c/wcag2ict#5]
(state of VoiceOver/Talkback turned on)In iOS explicitly provides a "Cancel" button to close the layer pop-up, Android does not have an explicit way to close the layer pop-up. However, the user can close the layer pop-up with a gesture (two-finger-tap). If gestures are not considered it should be developer to create a completely custom implementation? Or is gesture a mechanism for cancellation? I think this issue is extension of w3c/wcag#1070

iOS:
![C2613CDF-0F30-4863-B3FF-76690020B6CA](https://user-images.githubusercontent.com/7776510/90999129-be944d00-e600-11ea-84ce-1cccfdfaa127.png)
Android:
![Screenshot_20200824-103208](https://user-images.githubusercontent.com/7776510/90999135-c18f3d80-e600-11ea-8aa8-c512472e4858.png)

**Comment by maryjom** [comment]
I'm commenting regarding WCAG2ICT interpretation and guidance for non-web software and documents, since this had that label added. 

First off, since this issue was opened years ago, I'm not sure if this is still an issue or question for current Android devices. Does anyone know? I don't own an Android device to check.

For the 2.1.1 Keyboard accessibility requirement, if a keyboard can be attached to an Android device, then the pop-up should be dismissable via keyboard. From a WCAG2ICT perspective, is there any issue with non-web native software running on an Android device for dismissing pop-ups? Per current [WCAG2ICT guidance for 2.1.1](https://w3c.github.io/wcag2ict/#applying-sc-2-1-1-keyboard-to-non-web-documents-and-software), should still need to meet this requirement whenever a keyboard is supported.

For 2.5.1 Pointer Gestures, if there is a multi-point gesture that is used to dismiss the pop-up then to meet this requirement there needs to be a way to accomplish this using a single pointer. If the OS supports some mechanism for this to happen (even if it's using an accessibility setting or mode), then it would easily be met. If you are using native controls provided by the operating system, and there's an issue where single-pointer operation is not supported, the author would not be required to fix it. Instead, an OS bug should be opened to have the native OS solve the issue.

**Comment by mitchellevan** [comment]
Thank you @lgnacio for raising these good questions.

Speaking as a member of the WCAG2ICT task force, and an everyday Android user (without AT), and a semi-regular accessibility auditor on Android (with and without AT) — I agree that @patrickhlauke's and @maryjom's replies are correct and address the questions raised. 

As a result of this discussion I do not see any changes necessary for the WCAG2ICT Working Group Note. Therefore, @maryjom I recommend that this issue be closed.

**Comment by GreggVan** [comment]
> > Here is an example(Google map app)
> 
> that looks like a good approach/implementation to me
> 
> > Blind people are not recognize that relationships(layer and main screen) when the presentation format changes. So is it to fail the 1.3.1 A SC
> 
> I would say it's not necessary for a user to understand or not that they're in a layer over the main screen. at that point, they've jumped into a view that represents a menu. It's irrelevant if, in practice, that menu is "over" the main screen, or if it's screen-filling and effectively a separate screen, or anything else. The user's context at that point is "you're in this menu of options". So I don't see any issue here.

I agree as long as they are notified the they have changed context.

**Comment by GreggVan** [comment]
> Thank you @lgnacio for raising these good questions.
> 
> Speaking as a member of the WCAG2ICT task force, and an everyday Android user (without AT), and a semi-regular accessibility auditor on Android (with and without AT) — I agree that @patrickhlauke's and @maryjom's replies are correct and address the questions raised.
> 
> As a result of this discussion I do not see any changes necessary for the WCAG2ICT Working Group Note. Therefore, @maryjom I recommend that this issue be closed.

Not sure what you mean @mitchellevan.   @maryjom asked questions - so not sure what agreeing means.      I think the germs of an answer are in those posts (if MJs questions are answered yes) but someone has to propose the actual language of the reply - before we can close I think

**Comment by maryjom** [comment]
Draft TF Answer:
Since this issue was opened years ago, It seems this is no longer an issue for current Android devices. 

For the 2.1.1 Keyboard accessibility requirement, if a keyboard can be attached to an Android device, then the pop-up should be dismissable via keyboard. From a WCAG2ICT perspective, and per the current [WCAG2ICT guidance for 2.1.1](https://w3c.github.io/wcag2ict/#applying-sc-2-1-1-keyboard-to-non-web-documents-and-software), a native mobile application should still need to meet this requirement whenever a keyboard is supported.

For 2.5.1 Pointer Gestures, if there is a multi-point gesture that is used to dismiss the pop-up then to meet this requirement there would need to be a way to accomplish this using a single pointer. If the OS supports some mechanism for this to happen (even if it's using an accessibility setting or mode), then it would easily be met. If you are using native controls provided by the operating system, and there's an issue where single-pointer operation is not supported, the author would not be required to fix it. Instead, an OS bug should be opened to have the native OS solve the issue.

Therefore, the questions raised in this issue have all been addressed. If you disagree, open a new issue with examples the Task force can reproduce and examine.

**Comment by maryjom** [comment]
This issue is being closed as answered. Reopen this issue or open a new one if not resolved to your satisfaction.
